sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentPatchResourceIntegrationTest.java

```diff
@@ -0,0 +1,626 @@
+package io.stargate.sgv2.docsapi.api.v2.namespaces.collections.documents;
+
+import static io.restassured.RestAssured.given;
+import static net.javacrumbs.jsonunit.JsonMatchers.jsonEquals;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.notNullValue;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.TestProfile;
+import io.restassured.RestAssured;
+import io.stargate.sgv2.common.cql.builder.Replication;
+import io.stargate.sgv2.docsapi.config.constants.Constants;
+import io.stargate.sgv2.docsapi.service.schema.NamespaceManager;
+import io.stargate.sgv2.docsapi.service.schema.TableManager;
+import io.stargate.sgv2.docsapi.testprofiles.IntegrationTestProfile;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import javax.enterprise.context.control.ActivateRequestContext;
+import javax.inject.Inject;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.testcontainers.shaded.org.awaitility.Awaitility;
+
+@QuarkusTest
+@TestProfile(IntegrationTestProfile.class)
+@ActivateRequestContext
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class DocumentPatchResourceIntegrationTest {
+
+  public static final String BASE_PATH = "/v2/namespaces/{namespace}/collections/{collection}";
+  public static final String DEFAULT_NAMESPACE = RandomStringUtils.randomAlphanumeric(16);
+  public static final String DEFAULT_COLLECTION = RandomStringUtils.randomAlphanumeric(16);
+  public static final String DEFAULT_PAYLOAD =
+      "{\"test\": \"document\", \"this\": [\"is\", 1, true]}";
+  public static final String MALFORMED_PAYLOAD = "{\"malformed\": ";
+
+  @Inject NamespaceManager namespaceManager;
+
+  @Inject TableManager tableManager;
+
+  @Inject ObjectMapper objectMapper;
+
+  @BeforeAll
+  public void init() {
+
+    RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
+
+    namespaceManager
+        .createNamespace(DEFAULT_NAMESPACE, Replication.simpleStrategy(1))
+        .await()
+        .atMost(Duration.ofSeconds(10));
+
+    tableManager
+        .createCollectionTable(DEFAULT_NAMESPACE, DEFAULT_COLLECTION)
+        .await()
+        .atMost(Duration.ofSeconds(10));
+  }
+
+  @Nested
+  class PatchDocument {
+    @Test
+    public void happyPath() {
+      given()
+          .header(Constants.AUTHENTICATION_TOKEN_HEADER_NAME, "")
+          .header("Content-Type", "application/json")
+          .body(DEFAULT_PAYLOAD)
+          .when()
+          .patch(BASE_PATH + "/newdoc", DEFAULT_NAMESPACE, DEFAULT_COLLECTION)
```

Comment:
can you please use random doc ids everywhere, so that there is no chance of mixup in the tests

and also the call in the way

```
.patch(BASE_PATH + "/{document-id}", DEFAULT_NAMESPACE, DEFAULT_COLLECTION, id)
```